File: VB/SchedulingDemo.Wpf/ViewModels/OutlookInspired/ViewSettingsViewModel.vb
Imports System.Windows.Controls
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm.ViewModel

Namespace SchedulingDemo.ViewModels
    Public Class ViewSettingsViewModel
        Public Shared Function Create() As ViewSettingsViewModel
            Return ViewModelSource.Create(Function() New ViewSettingsViewModel())
        End Function

        Protected Sub New()
            ResetView()
        End Sub

        Public Overridable Property Orientation() As Orientation
        Public Overridable Property IsDataPaneVisible() As Boolean
        Public Sub ResetView()
            Orientation = Orientation.Horizontal
            IsDataPaneVisible = False
        End Sub

        Public Sub DataPaneRight()
            Orientation = Orientation.Horizontal
            IsDataPaneVisible = True
        End Sub
        Public Function CanDataPaneRight() As Boolean
            Return Orientation <> Orientation.Horizontal OrElse Not IsDataPaneVisible
        End Function
        Public Sub DataPaneBottom()
            Orientation = Orientation.Vertical
            IsDataPaneVisible = True
        End Sub
        Public Function CanDataPaneBottom() As Boolean
            Return Orientation <> Orientation.Vertical OrElse Not IsDataPaneVisible
        End Function
        Public Sub DataPaneOff()
            IsDataPaneVisible = False
        End Sub
        Public Function CanDataPaneOff() As Boolean
            Return IsDataPaneVisible
        End Function
    End Class
End Namespace

File: CS/SchedulingDemo.Wpf/ViewModels/OutlookInspired/ViewSettingsViewModel.cs
using System.Windows.Controls;
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm.ViewModel;

namespace SchedulingDemo.ViewModels {
    public class ViewSettingsViewModel {
        public static ViewSettingsViewModel Create() {
            return ViewModelSource.Create(() => new ViewSettingsViewModel());
        }

        protected ViewSettingsViewModel() {
            ResetView();
        }

        public virtual Orientation Orientation { get; set; }
        public virtual bool IsDataPaneVisible { get; set; }
        public void ResetView() {
            Orientation = Orientation.Horizontal;
            IsDataPaneVisible = false;
        }
              
        public void DataPaneRight() {
            Orientation = Orientation.Horizontal;
            IsDataPaneVisible = true;
        }
        public bool CanDataPaneRight() {
            return Orientation != Orientation.Horizontal || !IsDataPaneVisible;
        }
        public void DataPaneBottom() {
            Orientation = Orientation.Vertical;
            IsDataPaneVisible = true;
        }
        public bool CanDataPaneBottom() {
            return Orientation != Orientation.Vertical || !IsDataPaneVisible;
        }
        public void DataPaneOff() {
            IsDataPaneVisible = false;
        }
        public bool CanDataPaneOff() {
            return IsDataPaneVisible;
        }
    }
}
